File: src/decorator/java/com/my/NotificationSenderDecorator.java
package src.decorator.java.com.my;

public abstract class NotificationSenderDecorator implements NotificationSender {
    private final NotificationSender sender;

    public NotificationSenderDecorator(NotificationSender sender) {
        this.sender = sender;
    }

    @Override
    public void sendNotification() {
        sender.sendNotification();
    }
}

File: src/decorator/python/NotificationSenderDecorator.py
from abc import abstractmethod

from src.decorator.python.NotificationSender import NotificationSender


class NotificationSenderDecorator(NotificationSender):
    sender: NotificationSender

    def __init__(self, sender: NotificationSender):
        self.sender = sender

    @abstractmethod
    def sendNotification(self):
        self.sender.sendNotification()
